src/wrapper/clap-juce-wrapper.cpp

```diff
@@ -1053,6 +1055,47 @@ class ClapJuceWrapper : public clap::helpers::Plugin<
         return Plugin::voiceInfoGet(info);
     }
 
+#if JUCE_VERSION >= 0x080005
+    bool implementsNoteName() const noexcept override
+    {
+        return true;
+    }
+
+    uint32_t noteNameCount() noexcept override
+    {
+        noteNameInfoCached.clear();
+
+        // The JUCE docs say that plugins need to be able to handle
+        // note name requests for notes above 127, but CLAP only supports
+        // notes in range [0, 127].
+        for (int key = 0; key < 128; ++key)
+        {
+            // JUCE expects MIDI channels in range [1, 16].
+            // CLAP expects MIDI channels in range [0, 15].
+            for (int channel = 0; channel < 16; ++channel)
+            {
+                const auto optionalNoteName = processor->getNameForMidiNoteNumber (key, channel + 1);
+                if (optionalNoteName.has_value())
+                    noteNameInfoCached.push_back ({ *optionalNoteName, (int16_t) key, (int16_t) channel });
+            }
+        }
+
+        return static_cast<uint32_t> (noteNameInfoCached.size());
+    }
+
+    bool noteNameGet(uint32_t index, clap_note_name *noteName) noexcept override
+    {
+        if (index >= noteNameInfoCached.size())
+            return false;
+
+        const auto& noteNameInfo = noteNameInfoCached[index];
+        noteNameInfo.name.copyToUTF8 (noteName->name, CLAP_NAME_SIZE);
+        noteName->key = noteNameInfo.key;
+        noteName->channel = noteNameInfo.channel;
+        noteName->port = -1;
+        return true;
+    }
```

Comment:
I guess it is since we are indexed and keep the channel in the list of pairs so cool.